File: app/src/main/java/com/giosis/util/qdrive/singapore/AlertDialogActivity.java
package com.giosis.util.qdrive.singapore;


import android.app.Activity;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import androidx.core.app.NotificationManagerCompat;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.Window;
import android.widget.Button;
import android.widget.TextView;

import com.giosis.util.qdrive.main.MainActivity;
import com.giosis.util.qdrive.util.DataUtil;

public class AlertDialogActivity extends Activity {

    private String notiMessage;
    private String notiTitle;
    private String actionKey;
    private String actionValue;

    private String outletPush = "N";

    @Override
    public void onCreate(Bundle savedInstanceState) {

        setFinishOnTouchOutside(false);

        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        setContentView(R.layout.dialog_push_alert);

        TextView text_dialog_title = findViewById(R.id.text_dialog_title);
        TextView text_dialog_message = findViewById(R.id.text_dialog_message);
        Button btn_dialog_ok = findViewById(R.id.btn_dialog_ok);


        Bundle bun = getIntent().getExtras();

        notiMessage = bun.getString("notiMessage");
        notiTitle = bun.getString("notiTitle");
        actionKey = bun.getString("actionKey");
        actionValue = bun.getString("actionValue");

        try {

            outletPush = bun.getString("outletPush");
        } catch (Exception e) {

            outletPush = "N";
        }

        if (!notiTitle.equals("") && !notiTitle.equals("QSign SG")) {

            text_dialog_title.setVisibility(View.VISIBLE);
            text_dialog_title.setText(notiTitle);
        } else {

            text_dialog_title.setVisibility(View.GONE);
        }

        text_dialog_message.setText(notiMessage);
        btn_dialog_ok.setOnClickListener(clickListener);
    }

    @Override
    public void onBackPressed() {
    }

    OnClickListener clickListener = new OnClickListener() {
        @Override
        public void onClick(View v) {

            if (outletPush != null && outletPush.equals("Y")) {

                Intent intent = new Intent(AlertDialogActivity.this, MainActivity.class);
                intent.putExtra("outletPush", outletPush);
                startActivity(intent);
                overridePendingTransition(0, 0);
            } else if (actionKey.equals("LAE")) {      // 2019.02

                try {
                    if (actionValue != null) {

                        // ok 버튼 눌러서 이동하면 Notification 지우기
                        NotificationManagerCompat.from(getApplication()).cancel(Integer.parseInt(actionValue.substring(0, 9)));
                    }
                } catch (Exception e) {

                }

                Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(DataUtil.locker_pin_url));
                startActivity(intent);
            }

            finish();
        }
    };
}

File: app/src/main/java/com/giosis/util/qdrive/singapore/AlertDialogActivity.kt
package com.giosis.util.qdrive.singapore

import android.annotation.SuppressLint
import android.app.Activity
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.View
import android.view.Window
import androidx.core.app.NotificationManagerCompat
import com.giosis.library.main.MainActivity
import com.giosis.util.qdrive.singapore.databinding.DialogPushAlertBinding
import com.giosis.library.util.DataUtil

class AlertDialogActivity : Activity() {

    private val binding by lazy {
        DialogPushAlertBinding.inflate(layoutInflater)
    }

    @SuppressLint("SetTextI18n")
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        setContentView(binding.root)

        setFinishOnTouchOutside(false)

        val bun = intent.extras

        if (bun != null) {
            val notiTitle = if (bun.getString(FCMIntentService.TITLE) != null) {
                bun.getString(FCMIntentService.TITLE)
            } else {
                ""
            }

            val notiMessage = if (bun.getString(FCMIntentService.MESSAGE) != null) {
                bun.getString(FCMIntentService.MESSAGE)
            } else {
                ""
            }

            val actionKey = if (bun.getString(FCMIntentService.ACTION_KEY) != null) {
                bun.getString(FCMIntentService.ACTION_KEY)
            } else {
                ""
            }

            val actionValue = if (bun.getString(FCMIntentService.ACTION_VALUE) != null) {
                bun.getString(FCMIntentService.ACTION_VALUE)
            } else {
                ""
            }


            if (actionKey == FCMIntentService.LZD_PICK) {
                binding.titleText.text = "Pickup no : $actionValue"
                binding.messageText.text = resources.getString(R.string.text_lazada_order) + "\n" + resources.getString(R.string.text_lazada_content)

            } else {
                if (notiTitle!!.isEmpty()) {
                    binding.titleText.visibility = View.GONE

                } else {
                    binding.titleText.visibility = View.VISIBLE
                    binding.titleText.text = notiTitle
                }

                binding.messageText.text = notiMessage
            }

            binding.okBtn.setOnClickListener {

                if (actionKey == FCMIntentService.SevenEle_TAKEBACK
                    || actionKey == FCMIntentService.FL_TAKEBACK
                    || actionKey == FCMIntentService.LZD_PICK
                ) {
                    val intent = Intent(this@AlertDialogActivity, MainActivity::class.java)
                    intent.putExtra(FCMIntentService.DOWNLOAD, "Y")
                    startActivity(intent)
                    overridePendingTransition(0, 0)

                } else if (actionKey == FCMIntentService.LAE) {      // 2019.02

                    if (actionValue!!.isEmpty()) {
                        try {
                            // ok 버튼 눌러서 이동하면 Notification 지우기
                            NotificationManagerCompat.from(application)
                                .cancel(actionValue.substring(0, 9).toInt())

                        } catch (e: Exception) {

                        }
                    }

                    val intent = Intent(Intent.ACTION_VIEW, Uri.parse(DataUtil.locker_pin_url))
                    startActivity(intent)

                } else if (actionKey == "LOGOUT") {
                    val intent = Intent(this@AlertDialogActivity, LoginActivity::class.java)
                    intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_SINGLE_TOP)
                    startActivity(intent)
                }
                finish()
            }
        }

    }

    override fun onBackPressed() {
        //back key 막기
    }

}